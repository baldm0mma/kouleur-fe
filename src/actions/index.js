export const setProjects = projects => ({
  type: 'SET_PROJECTS',
  projects
});

export const setPalettes = palettes => ({
  type: 'SET_PALETTES',
  palettes
});

export const setCurrentProject = (id, name) => ({
  type: 'SET_CURRENT_PROJECT',
  id,
  name
});

export const setError = error => ({
  type: 'SET_ERROR',
  error
});

export const toggleNewProject = boolean => ({
  type: 'TOGGLE_NEW_PROJECT',
  boolean
})
