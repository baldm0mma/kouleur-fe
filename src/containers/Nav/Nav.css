nav {
  background-color: black;
  color: #bbbbbb;
}
.nav-wrapper {
  display: flex;
  justify-content: space-between;
  width: 85%;
  margin: auto;
  padding: 1rem;
}

.nav-control-bar {
  display: flex;
}

#delete {
  align-self: flex-end;
  margin-bottom: 12px;
}

#delete:hover {
  transform: scale(1.3);
}

#new-project-dropdown {
  margin-top: 2rem;
}

#new-project-button {
  width: 15rem;
  align-self: flex-end;
}

.project-control-container {
  display: flex;
  flex-direction: column;
  padding: 1rem;
  font-size: 1.3rem;
}

.site-title {
  font-size: 5em;
  padding: 0;
}

.k {
  color: #368360;
}

.o {
  color: #357964;
}

.u {
  color: #356f69;
}

.l {
  color: #35656e;
}

.e {
  color: #355b73;
}

.u2 {
  color: #355178;
}

.r {
  color: #35477d;
}
