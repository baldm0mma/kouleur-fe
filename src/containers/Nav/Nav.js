import React from 'react'
import { connect } from 'react-redux'
import { Dropdown, Button, Icon } from 'semantic-ui-react'
import {setCurrentProject, toggleNewProject} from '../../actions'
import './Nav.css'


const Nav = ({ projects, setCurrentProject, toggleNewProject, currentProject }) => {

  const handleClick = (id, name) => {
    setCurrentProject(id, name)
  }

  const clickNewProject = () => {
    toggleNewProject(true)
  }

  const buildProjects = projects.map( proj => {
    return <Dropdown.Item text={proj.project_name}  id={proj.id} key={proj.id} onClick={() => handleClick(proj.id, proj.project_name)}/>
  })

  return(
    <nav>
      <section className='nav-wrapper'>
      <h1 className='site-title'>
        <span className='k'>K</span>
        <span className='o'>o</span>
        <span className='u'>u</span>
        <span className='l'>l</span>
        <span className='e'>e</span>
        <span className='u2'>u</span>
        <span className='r'>r</span>
        </h1>
      <div className='nav-control-bar'>
        <div className='project-control-container'>
          <Button id='new-project-button' onClick={clickNewProject}>New Project</Button>
          <Dropdown id='new-project-dropdown' text={`Project: ${currentProject.name}`}>
            <Dropdown.Menu>
              {buildProjects}
            </Dropdown.Menu>
          </Dropdown>
        </div>
        <Icon name='delete' id='delete' size='large' color='red'/>
      </div>
      </section>
    </nav>
  )

}


export const mapStateToProps = ({ projects, currentProject }) => ({
  projects,
  currentProject
})

export const mapDispatchToProps = dispatch => ({
  setCurrentProject: (id, name) => dispatch(setCurrentProject(id, name)),
  toggleNewProject: (boolean) => dispatch(toggleNewProject(boolean))

})


export default connect(mapStateToProps, mapDispatchToProps)(Nav)