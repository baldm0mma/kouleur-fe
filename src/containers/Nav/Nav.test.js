import React from 'react';
import { shallow } from 'enzyme';
import { mapDispatchToProps, mapStateToProps, Nav } from './Nav.js';
import { setProjects, setPalettes, setError, setCurrentProject, toggleNewProject } from '../../actions';

describe('Nav', () => {
  let currentProject
  let projects
  let wrapper

  beforeEach(() => {
    currentProject = {project_name: 'Project', id: 1}
    projects = [{project_name: 'Project', id: 1}]
    wrapper = shallow(<Nav  currentProject={currentProject} projects={projects}/>)
  })
  it('should match the snapshot', () => {
     expect(wrapper).toMatchSnapshot();
  })
   it('should map setPalletes', () => {
  const mockDispatch = jest.fn();
  const mockAction = setPalettes([{pallete1: 'fffff'}]);
  const mappedProps = mapDispatchToProps(mockDispatch);
  mappedProps.setPalettes([{pallete1: 'fffff'}]);
  expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  });
  it('should map setError', () => {
    const mockDispatch = jest.fn();
    const mockAction = setError('Error');
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setError('Error');
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  });
    it('should map setCurrentProject', () => {
    const mockDispatch = jest.fn();
    const mockAction = setCurrentProject({project_name: '1', id: 1});
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setCurrentProject({project_name: '1', id: 1});
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  });
    it('should map set projects', () => {
    const mockDispatch = jest.fn();
    const mockAction = setProjects([{project_name: '1', id: 1}]);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setProjects([{project_name: '1', id: 1}]);
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  });
    it('should map toggleNewProject', () => {
    const mockDispatch = jest.fn();
    const mockAction = toggleNewProject(false);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.toggleNewProject(false);
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);  
    })
    it('should map state to props', () => {
    const mappedProps = mapStateToProps(
      { currentProject: currentProject },
      { projects: projects }
    );
    expect(mappedProps).toEqual({ currentProject });
  });
  
  


})


