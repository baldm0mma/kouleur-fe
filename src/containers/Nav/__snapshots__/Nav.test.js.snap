// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `ShallowWrapper {}`;
